robotoff/slack.py

```diff
@@ -72,140 +123,72 @@ class SlackNotifier(SlackNotifierInterface):
         "en:nutriscore-grade-e",
     }
 
-    PRIVATE_MODERATION_LABELS = {
-        "face",
-        "head",
-        "selfie",
-        "hair",
-        "forehead",
-        "chin",
-        "cheek",
-        "tooth",
-        "eyebrow",
-        "ear",
-        "neck",
-        "jaw",
-        "nose",
-        "facial expression",
-        "glasses",
-        "eyewear",
-        "child",
-        "baby",
-        "human",
+    COLLAPSE_LINKS_PARAMS = {
+        "unfurl_links": False,
+        "unfurl_media": False,
     }
 
     def __init__(self, slack_token: str):
         """Should not be called directly, use the NotifierFactory instead."""
         self.slack_token = slack_token
 
-    def notify_image_flag(self, insights: List[RawInsight], source: str, barcode: str):
+    def notify_image_flag(
+        self, image_insights: List[RawInsight], source_image: str, barcode: str
+    ):
+        """Sends alerts to Slack channels for flagged images."""
         text = ""
         slack_channel: str = self.ROBOTOFF_PUBLIC_IMAGE_ALERT_CHANNEL
 
-        for insight in insights:
-            flag_type = insight.data["type"]
-            label = insight.data["label"]
+        for flagged in image_insights:
+            flag_type = flagged.data["type"]
+            label = flagged.data["label"]
+
+            if _sensitive_image(flag_type, label):
+                slack_channel = self.ROBOTOFF_PRIVATE_IMAGE_ALERT_CHANNEL
 
             if flag_type in ("safe_search_annotation", "label_annotation"):
-                if (
-                    flag_type == "label_annotation"
-                    and label in self.PRIVATE_MODERATION_LABELS
-                ) or flag_type == "safe_search_annotation":
-                    slack_channel = self.ROBOTOFF_PRIVATE_IMAGE_ALERT_CHANNEL
-
-                likelihood = insight.data["likelihood"]
-                text += "type: {}, label: {}, score: {}\n".format(
-                    flag_type, label, likelihood
-                )
+                likelihood = flagged.data["likelihood"]
+                text += f"type: {flag_type}\nlabel: *{label}*, score: {likelihood}\n"
             else:
-                match_text = insight.data["text"]
-                text += "type: {}, label: {}, match: {}\n".format(
-                    flag_type, label, match_text
-                )
+                match_text = flagged.data["text"]
+                text += f"type: {flag_type}\nlabel: *{label}*, match: {match_text}\n"
 
-        url = settings.OFF_IMAGE_BASE_URL + source
-        edit_url = "{}/cgi/product.pl?type=edit&code={}" "".format(
-            settings.BaseURLProvider().get(), barcode
-        )
-        text += url + "\n"
-        text += "edit: {}".format(edit_url)
+        edit_url = f"{settings.BaseURLProvider().get()}/cgi/product.pl?type=edit&code={barcode}"
+        image_url = settings.OFF_IMAGE_BASE_URL + source_image
```

Comment:
It seems we could squash https://github.com/openfoodfacts/robotoff/issues/412 by just adding a transformation here:
above:

```python
define low_res_image(source_image, version="400"):
    return source_image.replace(".jpg", "." + version + ".jpg")
```
here:

```python
        image_url = settings.OFF_IMAGE_BASE_URL + low_res_image(source_image)
```

The 400px version is auto generated by the server for every images.

Reply:
(you would have to also add a .jpg to your source_image in your tests)

Reply:
With the migration to the blocks API, we don't need to manually downscale the image - see[ the last message in the test channel](https://app.slack.com/client/T02KVRT1Q/C02P0PR5VS4).

For the image link, we probably want the full-scale image anyway. If the alert image looks suspicious from the Slack channel clicking on image link should add more detail to the image rather than show the same resolution as in the thumbnail.